plasmapy/utils/checks.py

```diff
@@ -103,6 +103,7 @@ def wrapper(*args, **kwargs):
                 can_be_complex = validation_settings.get('can_be_complex', False)
                 can_be_inf = validation_settings.get('can_be_inf', True)
                 can_be_nan = validation_settings.get('can_be_nan', False)
+                can_be_none = validation_settings.get('can_be_none', False)
```

Comment:
Somewhere else in the code, I used a keyword with a name like `none_shall_pass`, which I think is used similarly to `can_be_none` here.  It might be helpful to check for consistency between keywords, though I made a note for myself so I'll end up checking it at some point if you don't want to do it here.

The only reason why I prefer `none_shall_pass` is that whenever I read it, I think of Gandalf (as portrayed by Ian McKellen) saying it (though admittedly it's not an exact quote).   :dragon_face: :mountain: :ring: